sdk/compiler/damlc/daml-stdlib-src/DA/Crypto/Text.daml

```diff
@@ -97,6 +98,12 @@ isBytesHex hex = Text.length hex % 2 == 0 && isHex hex
 isDEREncoded : Text -> Bool
 isDEREncoded msg = isBytesHex msg && Text.substring 0 2 msg == "30" && Text.length msg == 2 * (2 + (hexByteToInt (Text.substring 2 2 msg)))
 
+-- | Computes the SHA256 hash of the decoded UTF8 bytes of the `Text`, and returns it in its hex-encoded
+-- form. The hex encoding uses lowercase letters.
+sha256 : BytesHex -> BytesHex
+sha256 msg | isBytesHex msg = primitive @"BESha256Hex" (Text.asciiToLower msg)
+sha256 msg = error $ "Message argument is not a hex string: " <> msg
+
```

Comment:
we need to add the feature to the compiler as well. and wrap this in something like
```

```suggestion
#ifdef DAML_CRYPTO_ADDITIONAL
-- | Computes the SHA256 hash of the decoded UTF8 bytes of the `Text`, and returns it in its hex-encoded
-- form. The hex encoding uses lowercase letters.
sha256 : BytesHex -> BytesHex
sha256 msg | isBytesHex msg = primitive @"BESha256Hex" (Text.asciiToLower msg)
sha256 msg = error $ "Message argument is not a hex string: " <> msg
#endif
```

Reply:
note: Taking into account that DAML_CRYPTO = 2.1, we could maybe simply remove it.

Reply:
Oh, great idea! That solves an issue I'd seen on the next PR as we fold these changes in.

I'll leave removing `DAML_CRYPTO` to that PR for simplicity.